node/tracing-logging-and-metrics-for-sdk-and-core.md

```diff
@@ -0,0 +1,96 @@
+- Gauge of in-flight Activities - Lang
+- Gauge of in-flight WF activations - Lang
+- GRPC call latency and count per status for each route - Core only (can we just use otel spans?)
+
+> PS: More metrics should be considered here, this is just a list off the top of my head.<br/>
+> PPS: We might not need metrics from Core if we can rely on otel spans since those can be converted into metrics using a custom propagator.
+
+#### Tracing
+
+- Starting from Lang
+  - Poll start to completion (WFs and Activities)
+    - Lang passes span context to Core
+  - Connect to Server?
+- Starting from Core
+  - TODO
+
+#### Logging
+
+- Worker logger supports 4 log levels: `DEBUG | INFO | WARN | ERROR`.
+- Logging in trace context can be done with otel span events or by extracting the span context into a log event, it's unclear to me if otel spans are a good fit here, see [this discussion](https://github.com/open-telemetry/opentelemetry-specification/issues/67) for info on the subject.
+
+| Level              | Component | Trace context | Event                                                 |
+| ------------------ | --------- | ------------- | ----------------------------------------------------- |
+| `INFO`             | Worker    | None          | Worker state change                                   |
+| `DEBUG`            | Worker    | Activation    | Activation received with list of jobs and WF context  |
+| `ERROR`            | Worker    | Activation    | WF activation failed                                  |
+| `DEBUG`            | Workflow  | Activation    | WF activation concluded with list of commands         |
+| `DEBUG`            | Worker    | Activity Task | Activity task received (start or cancel) with context |
+| `DEBUG` or `ERROR` | Worker    | Activity Task | Activity task resolved                                |
+| `ERROR`            | Worker    | Any           | Unexpected error occurred                             |
+| `ERROR`            | Workflow  | Activation    | Unexpected error occurred                             |
+| `DEBUG`            | Worker    | None          | Got activity heartbeat                                |
+
+> TODO: Fill in this table with required Core logs
+
+### Solution
+
+- Add context to logs emitted from Worker where applicable
+- Pass span context as from Lang to Core in the `poll_activity_task` and `poll_workflow_activation` APIs
+- Add methods to the Core API for exporting logs, metrics and traces
+  - Special care should be taken when defining the types here, logs should be structured to support custom formatting on the Lang side
+  - Hook up instrumentation into Lang opentelemetry tracer and Worker logger
+- Inject the Worker logger interface into the isolate for internal use - make sure activation context is baked into the WF logger
+- Worker logger should add a `component` attribute (e.g. `worker | core | workflow`) in the log context so we know each log's origin
```

Comment:
If `workflow` is a type then `worker` should probably just be `activity` or something

Reply:
`workflow` refers to the WF isolate runtime library, for which logs should collected by the worker.